### PR TITLE
lib/libc/semaphore/sem_init : Register to sem list only if sem is fro…

### PR DESCRIPTION
**Pull request by jeongchanKim**
…m kernel

check that sem is from kernel or not when register to the g_sem_list, because g_sem_list manages only the kernel sem.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [38b7874c47e9dc3b55ceb68277b42d02cab6b685] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [38b7874c47e9dc3b55ceb68277b42d02cab6b685] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [7e6bedfb30426baef47095d6c55e2e7e012b381d] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [7e6bedfb30426baef47095d6c55e2e7e012b381d] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [9c0801b8556b540a513674b2462fb653f0535f22] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [9c0801b8556b540a513674b2462fb653f0535f22] - Code Rule Check OK.

File: os/arch/arm/src/common/up_checkspace.c [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#ifdef CONFIG_BUILD_PROTECTED
+
+bool is_kernel_space(void *addr)
+{
+	uint32_t ksram_segment_end  = (uint32_t)__ksram_segment_start__  + (uint32_t)__ksram_segment_size__;
```
Comment (sunghan-chang) [inline os/arch/arm/src/common/up_checkspace.c]: stdint.h

File: os/arch/arm/src/common/up_checkspace.c [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	uint32_t kflash_segment_end = (uint32_t)__kflash_segment_start__ + (uint32_t)__kflash_segment_size__;
+
+	if ((addr >= (void *)__ksram_segment_start__ && addr <= (void *)ksram_segment_end) || (addr >= (void *)__kflash_segment_start__ && addr < (void *)kflash_segment_end)) {
+		return true;
```
Comment (sunghan-chang) [inline os/arch/arm/src/common/up_checkspace.c]: stdbool.h

File: os/include/tinyara/arch.h [hunk 3]
```diff
@@ -2148,6 +2148,17 @@ void up_wdog_keepalive(void);
...
 
+#ifdef CONFIG_BUILD_PROTECTED
+/****************************************************************************
+ * Name: is_kernel_space
+ *
+ * Description:
+ *   Check the address is in kernel space or not
+ *
+ ****************************************************************************/
+bool is_kernel_space(void *addr);
```
Comment (sunghan-chang) [inline os/include/tinyara/arch.h]: is there including stdbool.h?

**Comment by seinfra** [comment]
Target : [294e77b0b2954111d906f82a6495473c97b10da1] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [294e77b0b2954111d906f82a6495473c97b10da1] - Code Rule Check (C++) OK.

File: os/include/tinyara/arch.h (hunk 3 above)
Comment (jeongchanKim) [reply os/include/tinyara/arch.h]: Yes. Already there.

File: os/arch/arm/src/common/up_checkspace.c (hunk 1 above)
Comment (jeongchanKim) [reply os/arch/arm/src/common/up_checkspace.c]: Done. Thanks.

File: os/arch/arm/src/common/up_checkspace.c (hunk 2 above)
Comment (jeongchanKim) [reply os/arch/arm/src/common/up_checkspace.c]: Done. Thanks.

**Comment by seinfra** [comment]
Target : [ca4e66413123c578a63d9b8f777381b5f37feca1] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [ca4e66413123c578a63d9b8f777381b5f37feca1] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [b561a90a0d351e1fdda47e364afcb8c92edd17e7] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [b561a90a0d351e1fdda47e364afcb8c92edd17e7] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [4e732943013be63b55d451a1b1665221534ac330] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [4e732943013be63b55d451a1b1665221534ac330] - Code Rule Check OK.

**Pull request merged**